bug(tests): unable to run tests with uv run due to none working dir

## Issue Description

Running tests with uv run doesn't work anymore

### How to reproduce

1. Ensure the dependencies are up to date with `uv sync --group test --reinstall`
2. Run `uv run pytest -n auto  tests/`
3. See the following error

```terminaloutput
________________________________________________________________ ERROR at setup of TestAggregationWithNoneValues.test_variance_ignores_none ________________________________________________________________
[gw9] darwin -- Python 3.10.15 /Users/alessandropomponio/Documents/GitHub/public/ado/.venv/bin/python3

    @pytest.fixture(scope="session", autouse=True)
    def initialize_ray() -> Generator[None, None, None]:
        """Initialize Ray with working_dir=None to avoid package size issues during tests."""
        # Using dict form instead of RuntimeEnv object - they behave differently
>       ray.init(
            runtime_env={"working_dir": None},
            ignore_reinit_error=True,
        )

tests/conftest.py:58: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.venv/lib/python3.10/site-packages/ray/_private/client_mode_hook.py:107: in wrapper
    return func(*args, **kwargs)
.venv/lib/python3.10/site-packages/ray/_private/worker.py:1798: in init
    runtime_env = _maybe_modify_runtime_env(runtime_env, _skip_env_hook)
.venv/lib/python3.10/site-packages/ray/_private/worker.py:1396: in _maybe_modify_runtime_env
    return hook(runtime_env)
.venv/lib/python3.10/site-packages/ray/_private/runtime_env/uv_runtime_env_hook.py:411: in hook
    if _is_path(working_dir):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

path_or_uri = None

    def _is_path(path_or_uri: str) -> bool:
        """Returns True if uri_or_path is a path and False otherwise.
    
        IMPORTANT: This is a duplicate of ray._private.path_utils.is_path().
    
        Why we can't import from path_utils:
        - This hook runs via `uv run --no-project uv_runtime_env_hook.py` in test scenarios
        - UV creates a minimal environment without dependencies installed yet
        - Importing from ray._private.path_utils triggers the full Ray import chain:
          ray._private.path_utils → ray/__init__.py → ray._private.worker →
          ray.widgets → ray.widgets.util → packaging.version
        - The 'packaging' module is not available in the minimal UV environment,
          causing: ModuleNotFoundError: No module named 'packaging.version'
    
        This duplicate implementation uses only stdlib (pathlib, urllib.parse)
        to avoid the dependency issue. If you modify this function, ensure you
        also update ray._private.path_utils.is_path() to keep them in sync.
        """
        if not isinstance(path_or_uri, str):
>           raise TypeError(f"path_or_uri must be a string, got {type(path_or_uri)}.")
E           TypeError: path_or_uri must be a string, got <class 'NoneType'>.
```

## Expected behaviour

The test work as expected

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -61,6 +61,7 @@ dev = [
   "pytest>=8.3.5",
   "pytest-asyncio>=1.0.0",
   "pytest-cov>=7.0.0",
+  "pytest-env>=1.1.0",
   "pytest-recording>=0.13.4",
   "pytest-timeout>=2.4.0",
   "pytest-xdist>=3.6.1",
@@ -114,6 +115,10 @@ source = "uv-dynamic-versioning"
 
 [tool.hatch.build.targets.wheel]
 packages = ["orchestrator"]
+
+# Read by Ray on first `import ray` (pytest-env runs when pytest starts, before conftest).
+[tool.pytest_env]
+RAY_ENABLE_UV_RUN_RUNTIME_ENV = "0"
 
 [tool.uv]
 default-groups = ["dev", "docs"]

--- a/requirements.txt
+++ b/requirements.txt
@@ -422,6 +422,7 @@ googleapis-common-protos==1.74.0 \
     # via google-api-core
 greenlet==3.4.0 ; platform_machine == 'AMD64' or platform_machine == 'WIN32' or platform_machine == 'aarch64' or platform_machine == 'amd64' or platform_machine == 'ppc64le' or platform_machine == 'win32' or platform_machine == 'x86_64' \
     --hash=sha256:04403ac74fe295a361f650818de93be11b5038a78f49ccfb64d3b1be8fbf1267 \
+    --hash=sha256:0e1254cf0cbaa17b04320c3a78575f29f3c161ef38f59c977108f19ffddaf077 \
     --hash=sha256:1054c5a3c78e2ab599d452f23f7adafef55062a783a8e241d24f3b633ba6ff82 \
     --hash=sha256:16dec271460a9a2b154e3b1c2fa1050ce6280878430320e85e08c166772e3f97 \
     --hash=sha256:1a54a921561dd9518d31d2d3db4d7f80e589083063ab4d3e2e950756ef809e1a \
@@ -435,20 +436,27 @@ greenlet==3.4.0 ; platform_machine == 'AMD64' or platform_machine == 'WIN32' or 
     --hash=sha256:5b99e87be7eba788dd5b75ba1cde5639edffdec5f91fe0d734a249535ec3408c \
     --hash=sha256:5cb614ace7c27571270354e9c9f696554d073f8aa9319079dcba466bbdead711 \
     --hash=sha256:636d2f95c309e35f650e421c23297d5011716be15d966e6328b367c9fc513a82 \
+    --hash=sha256:6f0def07ec9a71d72315cf26c061aceee53b306c36ed38c35caba952ea1b319d \
     --hash=sha256:805bebb4945094acbab757d34d6e1098be6de8966009ab9ca54f06ff492def58 \
     --hash=sha256:8424683caf46eb0eb6f626cb95e008e8cc30d0cb675bdfa48200925c79b38a08 \
     --hash=sha256:849f8bc17acd6295fcb5de8e46d55cc0e52381c56eaf50a2afd258e97bc65940 \
+    --hash=sha256:89995ce5ddcd2896d89615116dd39b9703bfa0c07b583b85b89bf1b5d6eddf81 \
+    --hash=sha256:8c5696c42e6bb5cfb7c6ff4453789081c66b9b91f061e5e9367fa15792644e76 \
     --hash=sha256:90036ce224ed6fe75508c1907a77e4540176dcf0744473627785dd519c6f9996 \
     --hash=sha256:9390ad88b652b1903814eaabd629ca184db15e0eeb6fe8a390bbf8b9106ae15a \
     --hash=sha256:956215d5e355fffa7c021d168728321fd4d31fd730ac609b1653b450f6a4bc71 \
+    --hash=sha256:98eedd1803353daf1cd9ef23eef23eda5a4d22f99b1f998d273a8b78b70dd47f \
     --hash=sha256:9b2d9a138ffa0e306d0e2b72976d2fb10b97e690d40ab36a472acaab0838e2de \
     --hash=sha256:a0a53fb071531d003b075c444014ff8f8b1a9898d36bb88abd9ac7b3524648a2 \
     --hash=sha256:a19093fbad824ed7c0f355b5ff4214bffda5f1a7f35f29b31fcaa240cc0135ab \
     --hash=sha256:a1c4f6b453006efb8310affb2d132832e9bbb4fc01ce6df6b70d810d38f1f6dc \
     --hash=sha256:a70ed1cb0295bee1df57b63bf7f46b4e56a5c93709eea769c1fec1bb23a95875 \
+    --hash=sha256:ac6a5f618be581e1e0713aecec8e54093c235e5fa17d6d8eb7ffc487e2300508 \
     --hash=sha256:b45e45fe47a19051a396abb22e19e7836a59ee6c5a90f3be427343c37908d65b \
+    --hash=sha256:b7857e2202aae67bc5725e0c1f6403c20a8ff46094ece015e7d474f5f7020b55 \
     --hash=sha256:c660bce1940a1acae5f51f0a064f1bc785d07ea16efcb4bc708090afc4d69e83 \
     --hash=sha256:d18eae9a7fb0f499efcd146b8c9750a2e1f6e0e93b5a382b3481875354a430e6 \
+    --hash=sha256:d336d46878e486de7d9458653c722875547ac8d36a1cff9ffaf4a74a3c1f62eb \
     --hash=sha256:ee407d4d1ca9dc632265aee1c8732c4a2d60adff848057cdebfe5fe94eb2c8a2 \
     --hash=sha256:f38b81880ba28f232f1f675893a39cf7b6db25b31cc0a09bb50787ecf957e85e \
     --hash=sha256:f50a96b64dafd6169e595a5c56c9146ef80333e67d4476a65a9c55f400fc22ff \

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -53,10 +53,13 @@ def random_identifier() -> Callable[[], str]:
 
 @pytest.fixture(scope="session", autouse=True)
 def initialize_ray() -> Generator[None, None, None]:
-    """Initialize Ray with working_dir=None to avoid package size issues during tests."""
-    # Using dict form instead of RuntimeEnv object - they behave differently
+    """Start Ray for the test session.
+
+    For ``uv run pytest``, the Ray "uv run" driver hook must be off by ensuring
+    ``RAY_ENABLE_UV_RUN_RUNTIME_ENV`` when ``ray`` is first imported. This
+    is set for pytest via ``pytest-env`` in ``pyproject.toml``
+    """
     ray.init(
-        runtime_env={"working_dir": None},
         ignore_reinit_error=True,
     )
     yield

